File: Leetcode/python/sol455.py
#  https://leetcode.com/problems/assign-cookies/description/
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort(), s.sort()
        children = cookies = 0
        while cookies < len(s) and children < len(g):
            '''
            If statement check if the cookie is big enough to satisfy a kid
            if it's big enough, count the kid and move on to match the next cookie.
            '''
            if s[cookies ]> =g[children]:
                childre n+ =1
            cookie s+ =1
        return children

File: Leetcode/cpp/sol455.cpp

// https://leetcode.com/problems/assign-cookies/

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {

        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int children = 0, cookies = 0;

        while(children < g.size() && cookies < s.size()){
            if(s[cookies]>=g[children]){
                children++;
            }
            cookies++;
        }

        return children;
    }
};